### PR TITLE
Fix EZP-25010: SiteAccessLimitationType should only accept values from existing SiteAccesses

### PR DESCRIPTION
**Pull request by lolautruche**
> https://jira.ez.no/browse/EZP-25010

### CODE REVIEW

**Comment by lolautruche** [comment]
Review ping @andrerom @bdunogier @pspanja @glye

File: eZ/Publish/Core/settings/settings.yml [hunk 1]
```diff
@@ -8,6 +8,7 @@ parameters:
     binaryfile_storage_prefix: original
     storage_dir: var/ezdemo_site/storage
     io_root_dir: %storage_dir%
+    ezpublish.siteaccess.list: [test]
```
Comment (glye) [inline eZ/Publish/Core/settings/settings.yml]: Testing artifact? :) Or should this be here?

**Comment by glye** [comment]
+1 apart from comment

File: eZ/Publish/Core/settings/settings.yml (hunk 1 above)
Comment (lolautruche) [inline eZ/Publish/Core/settings/settings.yml]: This is needed for tests as I added the dependency in `roles.yml`.

**Comment by bdunogier** [comment]
+1

File: eZ/Publish/Core/Limitation/Tests/SiteAccessLimitationTypeTest.php [hunk 2]
```diff
@@ -42,9 +44,20 @@ public function providerForTestAcceptValue()
...
+                            sprintf('%u', crc32('fre')),
+                        ),
+                    )
+                ),
+            ),
+            array(
+                new SiteAccessLimitation(
+                    array(
+                        'limitationValues' => array(
+                            crc32('ezdemo_site'),
```
Comment (andrerom) [inline eZ/Publish/Core/Limitation/Tests/SiteAccessLimitationTypeTest.php]: why is this not using sprintf? To emulate 32bit mode from legacy?

**Comment by andrerom** [comment]
+1

File: eZ/Publish/Core/Limitation/Tests/SiteAccessLimitationTypeTest.php (hunk 2 above)
Comment (lolautruche) [inline eZ/Publish/Core/Limitation/Tests/SiteAccessLimitationTypeTest.php]: To ensure it's working both with string and integer.

**Pull request merged**